File: obligation-cordapp/java-source/src/main/java/net/corda/examples/obligation/ObligationPlugin.java
package net.corda.examples.obligation;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.google.common.collect.ImmutableList;
import com.google.common.collect.ImmutableMap;
import net.corda.core.messaging.CordaRPCOps;
import net.corda.webserver.services.WebServerPluginRegistry;

import java.util.List;
import java.util.Map;
import java.util.function.Function;

public class ObligationPlugin implements WebServerPluginRegistry {
    private final List<Function<CordaRPCOps, ?>> webApis = ImmutableList.of(ObligationApi::new);

    private final Map<String, String> staticServeDirs = ImmutableMap.of(
            "obligation", getClass().getClassLoader().getResource("obligationWeb").toExternalForm()
    );

    @Override
    public List<Function<CordaRPCOps, ?>> getWebApis() {
        return webApis;
    }

    @Override
    public Map<String, String> getStaticServeDirs() {
        return staticServeDirs;
    }

    @Override
    public void customizeJSONSerialization(ObjectMapper objectMapper) {
    }
}

File: cordapp-upgrades/v2-finality-upgrade/src/main/kotlin/net/corda/examples/obligation/ObligationPlugin.kt
package net.corda.examples.obligation

import net.corda.core.messaging.CordaRPCOps
import net.corda.webserver.services.WebServerPluginRegistry
import java.util.function.Function

class ObligationPlugin : WebServerPluginRegistry {
    override val webApis: List<Function<CordaRPCOps, out Any>> = listOf(Function(::ObligationApi))
    override val staticServeDirs: Map<String, String> = mapOf(
            "obligation" to javaClass.classLoader.getResource("obligationWeb").toExternalForm()
    )
}